### ISSUE TITLE
Trouble using `nvim_call_atomic`

### ISSUE THREAD

**Issue by KillTheMule** [daa84/neovim-lib#14]
TL;DR: One needs to read the neovim documentation carefully, sorry for the noise!

Hey!

Could you help me figure out a problem? I'm not sure on which side it is, exactly :)

I've written the following function:

```Rust
pub fn resend_all_atomic(&self, nvim: &mut Neovim) -> Result<(), Error> {
  let mut calls: Vec<Value> = Vec::with_capacity(self.folds.len() + 1);

  calls.push(
    vec![
      Value::from("nvim_command".to_owned()),
      Value::from("normal! zE".to_owned()),
    ].into(),
  );

  for range in self.folds.keys() {
    calls.push(
      vec![
        Value::from("nvim_command".to_owned()),
        Value::from(format!("{},{}fo", range[0] + 1, range[1] + 1)),
      ].into(),
    );
  }

  nvim.call_atomic(calls).context("call_atomic failed")?;

  Ok(())
}

```

I compiles, so `calls` is indeed a `Vec<Value>`, and as can be seen, each `Value` is constructed from calling `into` on a `Vec<Value>`. So echo `Value` of `calls` is of variant `Array` with 2 elements. As far as I can see, that's the needed setup for neovim.

Still, neovim returns an error `1 - Args must be Array` (I assume 1 denotes the channel).

I've tried following through the code, but the only thing I really noticed is that `call_args!` wraps `calls`  it in another `Vec`, which is then passed down the chain, made part of an `RpcMessage` which is given to `model::encode`, where `rpc_args!` wraps it in another `Vec` together with the rpc metadata.

So what I'm saying is, I'm at a loss and I'd appreciate some help. If nothing else, a confirmation that `call_atomic` works as expected would be helpfull. Thanks in advance!

**Comment by KillTheMule** [comment]
Oh wait, just now I think I have it figured out. The arguments to each individual command must be packed into an array as well. I kinda misread the neovim documentation. I'll close the issue if this turns out to be the case!
